core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -146,6 +146,9 @@ class GroupCoordinator(val brokerId: Int,
       if (!group.is(Empty) && (!group.protocolType.contains(protocolType) || !group.supportsProtocols(protocols.map(_._1).toSet))) {
         // if the new member does not support the group protocol, reject it
         responseCallback(joinError(memberId, Errors.INCONSISTENT_GROUP_PROTOCOL))
+      } else if (group.is(Empty) && protocols.isEmpty ) {
+        //reject if first member with empty group protocol
+        responseCallback(joinError(memberId, Errors.INCONSISTENT_GROUP_PROTOCOL))
```

Comment:
The error code seems less than ideal, though I'm not sure there's a better alternative. Maybe we can at least extend the description in `Errors` to cover this case?

Reply:
updated the description of INCONSISTENT_GROUP_PROTOCOL error